frontend/src/editor/share.tsx

```diff
@@ -0,0 +1,200 @@
+import { IoIosTrash } from 'react-icons/io';
+import { HiOutlineClipboardCopy } from 'react-icons/hi';
+import clsx from 'clsx';
+import { DialogSeparator } from '../components/dialog';
+
+type ShareToken = ReturnType<typeof useListShareTokens>['data'][0];
+
+function pad(number: number) {
+  if (number < 10) {
+    return '0' + number;
+  }
+  return number;
+}
+
+function formatDate(date: Date) {
+  return (
+    date.getFullYear() +
+    '-' +
+    pad(date.getMonth() + 1) +
+    '-' +
+    pad(date.getDate()) +
+    'T' +
+    pad(date.getHours()) +
+    ':' +
+    pad(date.getMinutes()) +
+    ':' +
+    pad(date.getSeconds())
+  );
+}
+
+function getInNDays(n: number) {
+  const tomorrow = new Date();
+  tomorrow.setDate(tomorrow.getDate() + n); // setDate automatically fixes the date if we go beyond the month
+  return tomorrow;
+}
+
+function getShareUrl(token: string) {
+  const url = new URL(window.location.href);
+  const params = new URLSearchParams(url.search);
+  params.set('share_token', token);
+  url.search = params.toString();
+  return url.toString();
+}
+
+export function ShareTokenTable({
+  shareTokens,
+  documentId,
+  mutateShareTokens,
+}: {
+  shareTokens: ShareToken[];
+  documentId: string;
+  mutateShareTokens: () => void;
+}): JSX.Element {
+  if (!shareTokens || shareTokens.length == 0) {
+    return <></>;
+  }
+  return (
+    <>
+      <DialogSeparator />
+      <div>
+        <h3>Existing Share Tokens</h3>
+        <table className="table-auto border-separate border-spacing-1 w-full">
+          <thead>
+            <tr>
+              <td>Name</td>
+              <td>Valid Until</td>
+              <td></td>
+            </tr>
+          </thead>
+          <tbody>
+            {shareTokens.map((token, i) => (
+              <tr key={token.id || i}>
+                <td>{token.name}</td>
+                <td>
+                  {token.valid_until ? new Date(token.valid_until).toLocaleString() : 'Unlimited'}
+                </td>
+                <td>
+                  <IconButton
+                    icon={IoIosTrash}
+                    label={'delete share token'}
+                    onClick={async () => {
+                      try {
+                        await deleteShareToken({ document_id: documentId, token_id: token.id });
+                      } catch (e) {
+                        /* empty */
+                      }
+                      mutateShareTokens();
+                    }}
+                  />
+                </td>
+              </tr>
+            ))}
+          </tbody>
+        </table>
+      </div>
+    </>
+  );
+}
+export function ShareModal({
+  onClose,
+  documentId,
+  ...props
+}: {
+  onClose: () => void;
+  documentId: string;
+} & Omit<ComponentProps<typeof Modal>, 'label'>) {
+  const [name, setName] = useState('Unnamed');
+  const [limitValidity, setLimitValidity] = useState(true);
```

Comment:
can I have the default be false here?

Reply:
I'd prefer it to be false, to nudge the user to set a limit (security by default). But I that is no strong opinion